### ISSUE TITLE
Do we need temporary directories?

### ISSUE THREAD

**Issue by containscafeine** [ansible/ansible-container#188]
Hi,

A docker compose file relies on relative paths, like in case of `build`, `volumes`, `env_file` directives. Since we copy the generated docker compose file to a temporary directory under `/tmp/`, the relative files also have to be copied. This operation could be time consuming in some cases.

Does it make sense to not use temporary directories and supply docker compose files as python objects or some such?

Thoughts? Pointers?

**Comment by chouseknecht** [comment]
We should not need to copy files. In docker/engine.py the orchestrate() method sets the project path to the project base_path + '/ansible'. It should find the files. If it's not, then we're doing something wrong.

```
project = project_from_options(self.base_path + '/ansible', options)
```

Do you have an example where this is not working?

**Comment by dustymabe** [comment]
one example where this is a problem is where docker compose files have the `build` directive set and a relative path to a build context directory specified. Right now ansible-container creates a temporary directory and then creates a docker-compose file in there and then does a `docker-compose up`: https://github.com/ansible/ansible-container/blob/f5f425aaf40aec4d948d629a1201c349c63931b3/container/docker/engine.py#L776-L786

It appears that even though the real CWD is in the right location, docker-compose actually takes the relative path to be relative to the specified docker compose file (which is in /tmp/) and not CWD. This seems to be further proven by the comments in `https://github.com/docker/compose/issues/3568`

It looks like if we want this to work we need to put the templated out docker-compose file into the same directory as the original container.yml file was.

**Comment by chouseknecht** [comment]
Or, since we're constructing the docker-compose anyway, we could inspect the build: directive and turn the context path into an absolute path.

**Comment by dustymabe** [comment]
> Or, since we're constructing the docker-compose anyway, we could inspect the build: directive and turn the context path into an absolute path.

Absolute paths probably won't work since only the CWD gets bind mounted inside the `ansible-container` container.

**Comment by chouseknecht** [comment]
No. We call the compose *up* method to start all of the containers, and that's where the build would occur. It would be handled by the compose module, and thus outside of the Ansible build container. In fact, the Ansible process inside the build container never looks at the container.yml file. So in fact, we could alter the container.yml file in the temp directory, making the context path an absolute path.
